### PR TITLE
msg_copy_to_print_buffer

### PR DESCRIPTION
**Pull request by ethteck**
and some wip work. thanks to @Unnunu for getting the match with the do-whiles and Chippy in discord for some of the other improvements

### CODE REVIEW

**Comment by BowserSlug** [comment]
🚀 This PR matches 1 function (+0.12%) on `us`.

File: src/msg.c [hunk 1]
```diff
@@ -739,21 +741,23 @@ void msg_copy_to_print_buffer(MessagePrintState* printer, s32 arg1, s32 arg2) {
                             printer->maxLinesPerPage = 3;
                         }
                         printer->delayFlags |= 1;
-                        printer->stateFlags |= 0x800800;
+                        printer->stateFlags |= MSG_STATE_FLAG_800000 | MSG_STATE_FLAG_800;
                         if (nextArg != 0xC3) {
```
Comment (HailSanta) [inline src/msg.c:745]: MSG_CHAR_UNK_C3

File: src/msg.c [hunk 2]
```diff
@@ -777,51 +781,55 @@ void msg_copy_to_print_buffer(MessagePrintState* printer, s32 arg1, s32 arg2) {
                         printer->windowSize.x = *srcBuf++;
                         printer->windowSize.y = *srcBuf++;
                         sfx_play_sound_with_params(SOUND_21C, 0, 0, 0);
-                        printer->windowState = 2;
+                        printer->windowState = MSG_WINDOW_STATE_OPENING;
                         printer->delayFlags |= 1;
```
Comment (HailSanta) [inline src/msg.c:785]: MSG_DELAY_FLAG_1

File: src/msg.c [hunk 3]
```diff
@@ -777,51 +781,55 @@ void msg_copy_to_print_buffer(MessagePrintState* printer, s32 arg1, s32 arg2) {
...
+                        } while (0);
                         break;
                     case MSG_STYLE_POSTCARD:
                         arg = *srcBuf++;
-                        printer->windowState = 2;
-                        printer->stateFlags |= 0x800;
+                        printer->windowState = MSG_WINDOW_STATE_OPENING;
+                        printer->stateFlags |= MSG_STATE_FLAG_800;
                         printer->delayFlags |= 1;
                         printer->letterBackgroundImg = heap_malloc(7875);
```
Comment (HailSanta) [inline src/msg.c:805]: This is sizeX * sizeY / 2 for postcard, perhaps include the header and use those?

**Review by HailSanta — APPROVED** [review]

**Pull request merged**